osh/hub/scripts/osh-resalloc

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for resalloc agent spawner daemon
+This should be run as exactly the same user and group as the webserver. For example, `apache`.
+"""
+
+import logging
+import os
+import socket
+import tempfile
+
+logger = logging.getLogger(__name__)
+
+# On AWS, public hostname is accessible through this command on the vm:
+# `curl http://169.254.169.254/latest/meta-data/public-hostname`
+# TODO: Shall we use that instead of using ip address?
+def get_hostname_from_ip(ip_address):
+    logger.debug(f"Getting hostname for {ip_address}")
+    hostname = socket.gethostbyaddr(ip_address)[0]
+    return hostname
+
+
+def create_worker_in_kobo(worker_name):
+    from kobo.hub import models
+    kobo_worker = models.Worker.create_worker(worker_name)
+    # Each virtual machine allocated through resalloc should get exactly one task
+    # and it should be deleted after that.
+    kobo_worker.max_load = 1
+    kobo_worker.arches.add(models.Arch.objects.get(name="x86_64"))
+    kobo_worker.channels.add(models.Channel.objects.get(name="default"))
+    kobo_worker.save()
+    logger.debug(f"Created kobo worker for hostname {worker_name}")
+    return kobo_worker
+
+
+def start_worker(kobo_worker):
```

Comment:
Alternatively, content of this function could be put into an ansible playbook which should be run here. This way we would avoid directly running shell commands.

Reply:
I created a subpackage  for `osh-worker-manager` anyways.